mc/Mongo-SDAM.package/MongoSDAMClient.class/instance/stop.st

```diff
@@ -0,0 +1,3 @@
+API
+stop
+	monitors valuesDo: [ :each | each isRunning ifTrue: [ each stop ] ]
```

Comment:
I saw a test failure with the ExternalSemaphore table overflowing and started to look around. The closest to a smoking gun I have is this code.

1.) I am not sure if we need locking/concurrency control. If highly concurrent "each isRunning" make be called concurrently to each start?
2.) each stop returns a future. What do you think of waiting for the completion? I manually called waitForCompletion but somehow that doesn't work. The stopCallbacks is not ran as expected or it is too late.

Reply:
That might be because of debugging but TKTService has a race condition:

```
stop
	| futureStop |
	stopRequested := true.
	futureStop := TKTFuture new.
	self onStoppedDo: [ :v | futureStop deploySuccess: self ].
	TKTConfiguration serviceManager removeService: self.
	^ futureStop
```

We should set stopRequested _after_ the stoppedCallback is set. This is more important when single stepping through the code. WDUT? Can you have a look and upstream it?